### PR TITLE
docs: macOS development environment setup

### PR DESCRIPTION
**Pull request by lnielsen**

### CODE REVIEW

File: docs/getting-started/development-environment.rst [hunk 1]
```diff
@@ -9,10 +9,161 @@
...
+
+macOS
+~~~~~
+We recommend that you install libraries and tools using the
+`Homebrew <https://brew.sh>`_ package manager. Homebrew allows you to install
+two types of packages - cask and normal.
+
+**Cask packages**
+
+Cask packages are usually UI applications that ends up en your Applications
```
Comment (ppanero) [inline docs/getting-started/development-environment.rst:31]: 🇪🇸 
```suggestion
Cask packages are usually UI applications that end up in your Applications
```

File: docs/getting-started/development-environment.rst [hunk 2]
```diff
@@ -9,10 +9,161 @@
...
+
+Here are some recommended cask packages:
+
+.. code-block:: console
+
+    discord            # app that we use for chat among developers
+    docker             # docker setup for macOS
+    google-chrome      # needed for some end-to-end tests
+    iterm2             # a better terminal than Terminal.app
+    postico            # UI for PostgreSQL
```
Comment (ppanero) [inline docs/getting-started/development-environment.rst:46]: Not to polute, so it's ok as it is. I use `DB browser for SQLite` and it works pretty fine for sqlite. https://sqlitebrowser.org

File: docs/getting-started/development-environment.rst [hunk 3]
```diff
@@ -9,10 +9,161 @@
...
+    spectacle          # organise windows with keyboard shortcuts
+    visual-studio-code # a text editor used by many Invenio developers
+
+**Normal packages**
+
+Normal packages are usually command line tools/libraries that ends up in
```
Comment (ppanero) [inline docs/getting-started/development-environment.rst:52]:
```suggestion
Normal packages are usually command line tools/libraries that end up in
```

File: docs/getting-started/development-environment.rst [hunk 4]
```diff
@@ -9,10 +9,161 @@
...
+    libmemcached
+    libpng
+    libtiff
+    libxml2
+    libxslt
+    readline
+    xz
+    zlib
+
+Following is CLI tools that are useful during development:
```
Comment (ppanero) [inline docs/getting-started/development-environment.rst:82]:
```suggestion
Following are CLI tools that are useful during development:
```

**Review by ppanero — APPROVED** [review]
🚀 Super nice guide! Just a quick thought: It assumes XCode/DeveloperTools are installed already? It wouldn't be the first time that:
a- I forgot to install and got random `*.c / *.h` errors
b- A system update did something at it needs a reset/reinstall

Not sure if my comment above fits in the scope though.

**Comment by lnielsen** [comment]
Regarding XCode, then actually getting a bit in doubt. I probably installed it before running Homebrew, however, Homebrew for sure took care of installing the xcode cli tools.

File: docs/getting-started/development-environment.rst (hunk 2 above)
Comment (lnielsen) [reply docs/getting-started/development-environment.rst:46]: Let's add it :-) Idea is that people can select what they want. Probably I should add only docker is really required
``db-browser-for-sqlite``

File: docs/getting-started/development-environment.rst [hunk 5]
```diff
@@ -9,10 +9,161 @@
...
+    gifify       # make short screen recordings for bug reports
+    git          # our version control system
+    hub          # extends git with github features
+
+General CLI tools:
+
+.. code-block:: console
+
+    htop            # a better top
+    tree            # pretty print a directory structur
```
Comment (lnielsen) [inline docs/getting-started/development-environment.rst:98]:
```suggestion
    tree            # pretty print a directory structure
```

File: docs/getting-started/development-environment.rst [hunk 6]
```diff
@@ -9,10 +9,161 @@
...
+.. code-block:: console
+
+    $ mkvirtualenv <name>
+    $ mkvirtualenv -p python3.7 <name>
+    $ workon <name>
+
+Linux
+~~~~~
+
+Want to write it? Contact us on the chat!
```
Comment (lnielsen) [inline docs/getting-started/development-environment.rst:166]: Anyone on Linux which can help with this part? 😄

**Pull request merged**